### PR TITLE
Add new runtime functions for handling dependencies when initializing class metadata

### PR DESCRIPTION
**Pull request by rjmccall**
This is part of the fix for the race condition identified in rdar://47549859.  The fix in #22381 is too naive because we cannot stage it in; instead we need to make a runtime-only change to add new runtime functions, then switch the compiler to use them only when the compiler entrypoints can be safely presumed to exist.

### CODE REVIEW

**Comment by rjmccall** [comment]
> It looks like you didn't add compatibility override slots for the new entry points in this variation.

I don't think there were compatibility override slots for the existing entry points.

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by jckarter** [comment]
Ah, you're right, I had assumed that they were because of the other changes. Nevermind. Aside from [this issue](https://github.com/apple/swift/pull/22381/files#r254041393) it LGTM.

File: include/swift/Runtime/RuntimeFunctions.def [hunk 1]
```diff
@@ -808,6 +808,34 @@ FUNCTION(UpdateClassMetadata,
               SizeTy->getPointerTo()),
          ATTRS(NoUnwind))
 
+// struct FieldInfo { size_t Size; size_t AlignMask; };
```
Comment (brentdax) [inline include/swift/Runtime/RuntimeFunctions.def]: Just a comment fix, but what is this "FieldInfo"? Is it supposed to be the "TypeLayout" type mentioned in the comment's signature?

Comment (rjmccall) [reply include/swift/Runtime/RuntimeFunctions.def]: Yes.  I just copied that comment from another declaration, but I can fix it.

**Comment by rjmccall** [comment]
Private testing shows that the actual runtime code works.

**Comment by rjmccall** [comment]
I'll do Brent's request in the compiler PR.

**Pull request merged**